reporters/teamcity.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Results formatter for -R teamcity
+ *
+ * Options:
+ *  prefix-url - prefix the url to the metric key
+ *  postfix-url - postfix the url to the metric key
+ *  prefix - explicit prefix to the metric key
+ *  postfix - explicit postfix to the metric key
+ */
+'use strict';
+
+module.exports = function(results, reporterOptions) {
+
+    var isMultiple = Array.isArray(results);
+
+    function createTeamCityMessage(metric, value) {
+        var key = metric;
+        if (reporterOptions.prefix)
+            key = reporterOptions.prefix + '.' + key;
+        if (reporterOptions.postfix)
+            key = key + '.' + reporterOptions.postfix;
+        if (reporterOptions['prefix-url'] === true)
+            key = results.getUrl() + '.' + key;
+        if (reporterOptions['postfix-url'] === true)
+            key = key + '.' + results.getUrl();
+
+        return "##teamcity[buildStatisticValue key='" + key + "' value='" + value + "']";
```

Comment:
Good point, I guess the URL (prefix/postfix value) could contain that character. Are there any metric names that contain this character as well?